### PR TITLE
Registration form validation without any library 

### PR DESCRIPTION
**Pull request by Deepak8717**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/atapas/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

A Registration form validation without any library 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@Deepak8717 Is it ready for review?

**Comment by Deepak8717** [comment]
Yes @atapas , I have gone through the code review guideline and comments on previous pull request. Please review

File: src/plays/registration-form/RegistrationForm.jsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    errors: {
+      name: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+    },
+  });
+
+  //input values after submit
+  const [storedValues, setStoredValues] = useState({
```
Comment (atapas) [inline src/plays/registration-form/RegistrationForm.jsx:30]: Why do you need to maintain two states? Why can't the value states itself can be leveraged when the form submits?

File: src/plays/registration-form/RegistrationForm.jsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    let counter = 0;
+    Object.entries(values).forEach((value, i) => {
+      //validate each field, increase counter if validation passed.
+      if (validate(value)) {
+        counter = counter + 1;
+      }
+    });
+
+    //if all fields are validated succsfully store the values in new object and clear the input values
+    if (counter === 5) {
```
Comment (atapas) [inline src/plays/registration-form/RegistrationForm.jsx:143]: It may not be the best way to handle in a bigger scale application. So everytime there is a new field you need to change this condition to match with a new number. Can you think of a better way?

Note: I can accept with the current code but I think, it is ok to put more thoughts into it, what say?

**Review by atapas — COMMENTED** [review]
@Deepak8717 Comments....

**Comment by Deepak8717** [comment]
Thanks @atapas , These are great suggestion. There is always a room for improving. there is a lot more i could do like code enhancement, feature enhancement eg specific password/name format. show/hide error on input focus. I am always willing to push the limit but this project has really tested my js skills that i just wanted to finish to take a break and work on my other portfolio projects. May be return back after a week and raise new issues and enhance further.

**Comment by atapas** [comment]
> Thanks @atapas , These are great suggestion. There is always a room for improving. there is a lot more i could do like code enhancement, feature enhancement eg specific password/name format. show/hide error on input focus. I am always willing to push the limit but this project has really tested my js skills that i just wanted to finish to take a break and work on my other portfolio projects. May be return back after a week and raise new issues and enhance further.
> 
> tldr; improving everything in at one time difficult, may be take break and comeback with refresh mind always helps

I am fine as long as you keep improving :)

Kindly resolve the merge conflicts.. lets get this in.

Btw, I have seen that both the files have conflicts have been modified beyond the scope by you. Hence you got conflicts. You may want to check that out.

@Deepak8717

**Comment by Deepak8717** [comment]
@atapas merge resolved. Please check and accept the changes for time being

**Review by atapas — APPROVED** [review]
